charts/redpanda/chart_test.go

```diff
@@ -792,12 +803,19 @@ func TestGoHelmEquivalence(t *testing.T) {
 	helmObjs[stsIdx].(*appsv1.StatefulSet).Spec.Template.Spec.Containers[0].Resources, err = valuesutil.UnmarshalInto[corev1.ResourceRequirements](helmObjs[stsIdx].(*appsv1.StatefulSet).Spec.Template.Spec.Containers[0].Resources)
 	require.NoError(t, err)
 
+	// Drop annotations that gives different result as ConfigMap translated from go and helm template
+	// gives different result. The ConfigMap is first translated to yaml then calculated sha256sum.
+	goObjs[depIdx].(*appsv1.Deployment).Spec.Template.Annotations["checksum/config"] = ""
```

Comment:
We didn't need to do this anywhere else. Why do we need to do it here? Is this a divergence between the gocode and helm code?